src/main/scala/org/apache/mesos/chronos/scheduler/mesos/MesosTaskBuilder.scala

```diff
@@ -174,16 +174,51 @@ class MesosTaskBuilder @Inject()(val conf: SchedulerConfiguration) {
       v.mode.map { m =>
         volumeBuilder.setMode(Volume.Mode.valueOf(m.toString.toUpperCase))
       }
+      v.external.foreach { e =>
+        volumeBuilder.setSource(Volume.Source.newBuilder()
+            .setType(Volume.Source.Type.DOCKER_VOLUME)
+            .setDockerVolume(Volume.Source.DockerVolume.newBuilder()
+              .setDriver(e.provider)
+              .setName(e.name)
+              .setDriverOptions(Parameters.newBuilder()
+                .addAllParameter(e.options.map(_.toProto()).asJava).build()
+              ).build()
+            ).build()
+        ).build()
+      }
 
       volumeBuilder.build()
     }.foreach(builder.addVolumes)
-    builder.setType(ContainerInfo.Type.DOCKER)
-    builder.setDocker(DockerInfo.newBuilder()
-      .setImage(job.container.image)
-      .setNetwork(DockerInfo.Network.valueOf(job.container.network.toString.toUpperCase))
-      .setForcePullImage(job.container.forcePullImage)
-      .addAllParameters(job.container.parameters.map(_.toProto).asJava)
-      .build()).build
+
+    job.container.`type` match {
+      case ContainerType.DOCKER =>
+        builder.setType(ContainerInfo.Type.DOCKER)
+        builder.setDocker(DockerInfo.newBuilder()
+          .setImage(job.container.image)
+          .setNetwork(DockerInfo.Network.valueOf(job.container.network.toString.toUpperCase))
+          .setForcePullImage(job.container.forcePullImage)
+          .addAllParameters(job.container.parameters.map(_.toProto()).asJava)
+          .build())
+      case ContainerType.MESOS =>
+        builder.setType(ContainerInfo.Type.MESOS)
+        builder.setMesos(ContainerInfo.MesosInfo.newBuilder()
+          .setImage(Image.newBuilder()
+            // TODO add APPC image support
+            .setType(Image.Type.DOCKER)
+            .setDocker(Image.Docker.newBuilder()
+              .setName(job.container.image)
+              // TODO add setCredential
+              .build())
+            .setCached(!job.container.forcePullImage)
+            .build())
+          .build())
+    }
+    job.container.networkName.foreach {
+      n => builder.addNetworkInfos(NetworkInfo.newBuilder()
```

Comment:
I'm hesitant to accept this if it's incomplete, or if there's risk of API change. I'd prefer to have the API right from the start so we don't get 'stuck'.

Agreed on the last comment. I guess we should do something like this:

```json
{
  "container":{
    "type":"MESOS",
    "forcePullImage":true,
    "image":"debian",
    "networkInfos":{
      "name":"mynet"
    }
  }
}
```